flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/FixedDelayRestartBackoffTimeStrategy.java

```diff
@@ -0,0 +1,118 @@
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * Restart strategy which tries to restart a fixed number of times with a fixed backoff time in between.
+ */
+public class FixedDelayRestartBackoffTimeStrategy implements RestartBackoffTimeStrategy {
+
+	private final int maxNumberRestartAttempts;
+
+	private final long backoffTimeMS;
+
+	private final String strategyString;
+
+	private int currentRestartAttempt;
+
+	FixedDelayRestartBackoffTimeStrategy(int maxNumberRestartAttempts, long backoffTimeMS) {
+		checkArgument(maxNumberRestartAttempts >= 0, "Maximum number of restart attempts must be at least 0.");
+		checkArgument(backoffTimeMS >= 0, "Backoff time between restart attempts must be at least 0 ms.");
+
+		this.maxNumberRestartAttempts = maxNumberRestartAttempts;
+		this.backoffTimeMS = backoffTimeMS;
+		this.currentRestartAttempt = 0;
+		this.strategyString = generateStrategyString();
+	}
+
+	@Override
+	public boolean canRestart() {
+		return currentRestartAttempt <= maxNumberRestartAttempts;
+	}
+
+	@Override
+	public long getBackoffTime() {
+		return backoffTimeMS;
+	}
+
+	@Override
+	public void notifyFailure(Throwable cause) {
+		currentRestartAttempt++;
+	}
+
+	@Override
+	public String toString() {
+		return strategyString;
+	}
+
+	private String generateStrategyString() {
+		StringBuilder str = new StringBuilder("FixedDelayRestartBackoffTimeStrategy(");
+		str.append("maxNumberRestartAttempts=");
+		str.append(maxNumberRestartAttempts);
+		str.append(", backoffTimeMS=");
+		str.append(backoffTimeMS);
+		str.append(")");
+
+		return str.toString();
+	}
+
+	public static FixedDelayRestartBackoffTimeStrategyFactory createFactory(final Configuration configuration) {
+		return new FixedDelayRestartBackoffTimeStrategyFactory(configuration);
+	}
+
+	/**
+	 * The factory for creating {@link FixedDelayRestartBackoffTimeStrategy}.
+	 */
+	public static class FixedDelayRestartBackoffTimeStrategyFactory implements RestartBackoffTimeStrategy.Factory {
+
+		private final int maxNumberRestartAttempts;
+
+		private final long backoffTimeMS;
+
+		public FixedDelayRestartBackoffTimeStrategyFactory(final Configuration configuration) {
+			this.maxNumberRestartAttempts = configuration.getInteger(
+					RESTART_BACKOFF_TIME_STRATEGY_FIXED_DELAY_ATTEMPTS);
+			this.backoffTimeMS = getIntervalMSFromConfiguration(
+					configuration,
+					RESTART_BACKOFF_TIME_STRATEGY_FIXED_DELAY_BACKOFF_TIME);
+		}
+
+		@Override
+		public RestartBackoffTimeStrategy create() {
+			return new FixedDelayRestartBackoffTimeStrategy(
+					maxNumberRestartAttempts,
+					backoffTimeMS);
+		}
+
+		private long getIntervalMSFromConfiguration(Configuration configuration, ConfigOption<String> configOption) {
+			Duration interval = Duration.apply(configuration.getString(configOption));
+			return Time.milliseconds(interval.toMillis()).toMilliseconds();
+		}
```

Comment:
Duplicate logic wrt the `FailureRateRestartBackoffTimeStrategy`

Reply:
OK